### PR TITLE
#163518688  User Can Create Profile

### PR DESCRIPTION
**Pull request by YvesIraguha**
#### What does this PR do?
Enables the functionality of updating, fetching, and deleting the user profile. 
#### Description of Task to be completed?
* Create update profile functionality
* Create fetch user profile functionality 
* Create delete user profile functionality 
* Create tests for profile creation 
#### How should this be manually tested?
* Pull the remote branch using `git pull origin ft-profile-creation-163518688` 
* `cd project_directory`
* Run the database migrations by `sequelize db:migrate`
* Run tests using `yarn test` or
* Start the server using `yarn start`
* Sign up to the app by consuming `POST: /api/v1/users`
* Login to the app by consuming `POST: /api/v1/users/login`
* Grab the token and put in headers of the request like `{Authorization: Bearer <token> }`as you consume `PUT: /api/v1/profiles/:username` to update the profile. The body of the request should look like `req.body: { profile:{
	                               username:"YvesIraguha",
	                                 firstname:"yvesIraguhawuara",
	                                 bio: "I am a javascript developer"
                                          }
                                     }`

  

#### Any background context you want to provide?
After a user has signed up, he/she might want to update the profile. Therefore, these endpoints are for editing the profile created when a user signs up. 
#### What are the relevant pivotal tracker stories?
163518688

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -42,44 +46,37 @@ if (!isProduction) {
...
-});
+// app.use((err, res) => {
+//   res.status(err.status || 500);
+//   res.json({
+//     errors: {
+//       message: err.message,
+//       error: {}
+//     }
+//   });
+// });
```
Comment (niomwungeri-fabrice) [inline index.js]: Why did u comment out those lines of code @YvesIraguha

File: routes/profile.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import Profile from "../controllers/profileController";
+const profileRouter = Router();
+
+profileRouter
+  .put(
+    "/profiles/:username",
+    passport.authenticate("jwt", { session: false }),
+    Profile.update
+  )
+  .get("/profiles/:username", Profile.getProfile)
```
Comment (niomwungeri-fabrice) [inline routes/profile.js:12]: We need middleware to check whether the user has been identified for both of your routes @YvesIraguha `isVerified` true or false

Comment (YvesIraguha) [reply routes/profile.js:12]: Those routes, are not protected. There is no middleware needed there.

**Review by abayo-luc — APPROVED** [review]

**Review by Musigwa-zz — APPROVED** [review]
Looks great, it can be merged.

**Pull request merged**